### DISCUSSION TITLE
How to get active theme name in Vue3 and Nuxt3?

### DISCUSSION THREAD

**Discussion by mateoKutnjak** [saadeghi/daisyui#2454] [Q&A]
I am using `.env` file with content:

```
THEMES="corporate,night"
```

and my `tailwind.config.js` uses these variables to setup themes. 

```
daisyui: {
        base: false,
        themes: [...process.env.THEMES.split(','), {
            theme1: {
                "primary": "#d4be9b",
                "secondary": "#f6d860",
                "accent": "#824C71",
                "neutral": "#000001",
                "base-100": "#f6f6f6",
                "--rounded-btn": "0rem",
                "base-content": '#212121',
            },
            theme1: {
                "primary": "#C2384A",
                "secondary": "#f6d860",
                "accent": "#824C71",
                "neutral": "#000001",
                "base-100": "#101728",
                "base-200": '06080F',
                "--rounded-btn": "0rem"
            }
        }],
    },
```

Themes are loaded properly through `.env` file but first theme seems to be `night` which is the second theme in array. Is there any way to get name of the active theme because with regards to the theme I am changing background image and because `night` is the first theme that is set up I have a problem with wrong background. 

Thank you.

**Comment by saadeghi** [comment]
If the env variable is loaded properly it would work as expected and `corporate` would be the first theme:  
https://play.tailwindcss.com/DWyiPlsOn4?file=config

**Reply by mateoKutnjak** [reply]
But it is not a rule. Sometime it just starts off with dark theme, like its state is saved somewhere in cache. Is where a way to determine current theme?

**Reply by saadeghi** [reply]
>Sometime it just starts off with dark theme  

If it's not always working the same way, it's because of that env variable and the way your build tool handles env variables. It could be cache. I don't know.

>Is where a way to determine current theme?

Yes and no.

You should be able to read the `daisyui.theme` array from taiwind.config.js to see the order of themes. but daisyUI config can't see which one is currently active in runtime because the first theme is injected to `root`, the default dark theme gets injected to root with `prefers-color-scheme: dark`. the current theme might be light or dark based on OS settings. Other themes use `data-theme` and the theme inside `data-theme` will be the current theme:

https://unpkg.com/browse/daisyui@4.4.19/dist/themes.css

Let me know if you have a question.

**Comment by ghost** [comment]
For your reference:

```lang=typescript
<script lang="ts">
    import { writable, onMount } from 'svelte';
    import type { Writable } from 'svelte/store';
    import { themeChange } from "theme-change";
    // Assuming the existence of a 'themes' import

    // Step 1: Create a writable store
    const currentTheme: Writable<string> = writable('');

    onMount(() => {
        themeChange(false);

        // Initial update
        updateThemeStore();

        // Step 2: Set up a MutationObserver
        const observer: MutationObserver = new MutationObserver((mutations: MutationRecord[]) => {
            mutations.forEach((mutation: MutationRecord) => {
                if (mutation.type === 'attributes' && mutation.attributeName === 'data-theme') {
                    updateThemeStore();
                }
            });
        });

        observer.observe(document.documentElement, { attributes: true });

        return () => {
            observer.disconnect();
        };
    });

    // Function to update the theme store
    function updateThemeStore(): void {
        const theme: string | null = document.documentElement.dataset.theme;
        currentTheme.set(theme ?? 'default-theme');
    }

    // ... rest of your script
</script>

<!-- ... your HTML ... -->
```

**Comment by ghost** [comment]
Hi @saadeghi, thanks for your latest response. 

Could you let me know what caveat `document.documentElement.dataset.theme` will carry? I am currently using javascript/typescript to retrieve theme from this variable. I am a little bit confused on your last response when you said "yes and no". If my understanding is correct, DaisyUI is unaware of the current active theme? Does this mean `document.documentElement.dataset.theme` is not always reliable?

Thank you.

**Comment by saadeghi** [comment]
Let's say you don't set any `data-theme` attribute to the root element at all. In this case, the theme depends on the system (If OS is dark, it will be dark)

If you set `data-theme` attribute to the root element, then that will override the system theme and you can get the current active theme from `documentElement` using JS.

Let me know if you have a question.
